Google Cloud Providers - Fix _MethodDefault deepcopy failure

IKholopov opened the pull request:
This is the attempt to fix #28751 by setting a memo to the same instance of DEFAULT, which is the stub for a Literal value type introduced to support Python 3.7 with mypy in Google's API core Python client.

closes: #28751

eladkal commented on airflow/providers/google/common/operators/base_google.py:
```diff
@@ -0,0 +1,34 @@
+    """
+    Abstract base class that takes care of common specifics of the operators built
+    on top of Google API client libraries.
+    """
```
Our goal is to split Google provider to smaller providers https://github.com/apache/airflow/issues/15933
This class seems to be GCP specific so lets keep it in the `cloud` directory and not in `common` as this directory will not stay after the split.

Possibly also rename the class to`GcpBaseOperator` ?

IKholopov replied:
Good idea, especially since AFAIU only Cloud providers are affected. GCP is a bit of a deprecated term, but "Google Cloud Base" should do that.

potiuk commented on airflow/providers/google/cloud/operators/cloud_base.py:
```diff
@@ -0,0 +1,34 @@
+    """
+
+    def __deepcopy__(self, memo):
+        memo[id(DEFAULT)] = DEFAULT
```
Would be nice to add explanation why we are doing it.

potiuk replied:
(Link to some description/issue explaining it because otherwise it is pretty magical.

IKholopov replied:
Done.

Commit:
Google Cloud Providers - Fix _MethodDefault deepcopy failure

This is the attempt to fix #28751 by setting a memo to the same instance of DEFAULT,
which is the stub for a Literal value type introduced to support Python 3.7 with mypy
in Google's API core Python client.

Without this any operator that has the parameter set to DEFAULT constant
(which is often the default value for retry parameters) will throw the
error in mini-scheduler after execution as the attemt to deepcopy this
project would fail.
